File: app/src/main/java/de/htw/berlin/s0558606/lasersensorcommunicator/model/AppDatabase.java
package de.htw.berlin.s0558606.lasersensorcommunicator.model;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.content.Context;

/**
 * Created by Marcel Ebert S0558606 on 26.05.18.
 */
@Database(version = 1, entities = {MeasuringLocation.class, SensorData.class, Measurement.class})
public abstract class AppDatabase extends RoomDatabase {

    public static final String DATABASE_NAME = "data.db";

    private static AppDatabase sInstance;

    abstract public MeasuringLocationDao locationDao();

    abstract public SensorDataDao sensorDataDao();

    abstract public MeasurementDao measurementDao();

    public static AppDatabase getInstance(Context context) {
        if (sInstance == null) {
            synchronized (AppDatabase.class) {
                if (sInstance == null) {
                    sInstance = Room.databaseBuilder(context.getApplicationContext(),
                            AppDatabase.class, DATABASE_NAME).allowMainThreadQueries()
                            .build();
                }
            }
        }
        return sInstance;
    }

}

File: app/src/main/java/de/htw/berlin/s0558606/lasersensorcommunicator/model/AppDataBase.kt
package de.htw.berlin.s0558606.lasersensorcommunicator.model

import android.arch.persistence.room.Database
import android.arch.persistence.room.Room
import android.arch.persistence.room.RoomDatabase
import android.content.Context


@Database(entities = arrayOf(Location::class, SensorData::class), version = 5)
abstract class AppDatabase : RoomDatabase() {

    /*//////////////////////////////////////////////////////////
    // ABSTRACT METHODS
    *///////////////////////////////////////////////////////////
    abstract fun locationDao(): LocationDao
    abstract fun sensorDataDao(): SensorDataDao


    /*//////////////////////////////////////////////////////////
    // COMPANION OBJECT
    *///////////////////////////////////////////////////////////
    companion object {
        val DB_NAME = "data_db"
        var dbInstance: AppDatabase? = null

        fun getDatabase(context: Context) : AppDatabase? {
            if (dbInstance == null) {
                dbInstance = Room.databaseBuilder<AppDatabase>(context.applicationContext, AppDatabase::class.java, DB_NAME).build()
            }
            return dbInstance
        }
    }

}